compute_endpoint/globus_compute_endpoint/engines/process_pool.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+import logging
+import multiprocessing
+import queue
+import typing as t
+import uuid
+from concurrent.futures import Future
+from concurrent.futures import ProcessPoolExecutor as NativeExecutor
+
+import psutil
+from globus_compute_common.messagepack.message_types import (
+    EPStatusReport,
+    TaskTransition,
+)
+from globus_compute_endpoint.engines.base import (
+    GlobusComputeEngineBase,
+    ReportingThread,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ProcessPoolEngine(GlobusComputeEngineBase):
+    def __init__(
+        self,
+        *args,
+        label: str = "ProcessPoolEngine",
+        heartbeat_period_s: float = 30.0,
+        **kwargs,
+    ):
+        self.label = label
+        self._executor = None
+        self._executor_args = args
+        self._executor_kwargs = kwargs
+        self._status_report_thread = ReportingThread(
+            target=self.report_status, args=[], reporting_period=heartbeat_period_s
+        )
+        super().__init__(*args, heartbeat_period_s=heartbeat_period_s, **kwargs)
+
+    @property
+    def executor(self):
```

Comment:
Summarizing notes from a PR discussion:
Since the executor is not explicitly defined unless it is called, it could be started either by the user calling `submit` or by the `_status_report_thread`, which is awkward and could potentially hide errors at instantiation. It'd be nice to define the executor in the start method, and have asserts in both `get_status_report` and `submit` in case they are used before `start`.